executor/load_data.go

```diff
@@ -267,7 +267,13 @@ func (e *LoadDataInfo) colsToRow(cols []string) types.DatumRow {
 			e.row[i].SetString("")
 			continue
 		}
-		e.row[i].SetString(cols[i])
+		// "\N" is handled as NULL in csv.
+		// See http://dev.mysql.com/doc/refman/5.7/en/load-data.html
+		if cols[i] == "\\N" {
+			e.row[i].SetNull()
```

Comment:
You need to check e.FieldsInfo.Enclosed otherwise the following would be foo,<nil> (want foo,\N) if enclosed with "
```
"foo","\\N"
```